test/neural_net/style_transfer_testing.hpp

```diff
@@ -0,0 +1,63 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#ifndef __TEST_NEURAL_NET_STYLE_TRANSFER_TESTING_HPP
+#define __TEST_NEURAL_NET_STYLE_TRANSFER_TESTING_HPP
+
+#include <functional>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace neural_net_test {
+namespace style_transfer {
+
+class BaseNetworkTest {
+ public:
+  BaseNetworkTest() = delete;
+  explicit BaseNetworkTest(float epsilon);
+
+  ~BaseNetworkTest();
+
+  /**
+    Checks the encoding prediction by using an input configuration and an input
+    bin file. The configuration file has 3 major keys:
```

Comment:
What is a "bin file"? What is a "configuration file?" Might need some higher-level context here

Reply:
The bin file contains the data, the json files contain the format in which the data is stored. For example let's take an array as an input.

The array has a shape, and data. The shape information would go in the config file, whereas the actual bytes would be stored in the bin file. 

These arrays could be an inputs or outputs. 

There could be multiple array's (multiple inputs and outputs) as well so bin and configuration files seemed to be the easiest way to go for testing. Probably not ideal for production.

Earlier I had a JSON file containing both the format data and the actual data. That significantly slowed my testing process since I was directly compiling this json file into the test executable and parsing the json for the weights. Both compile time and run-time were very slow using this method.

A potential solution to this problem:
- Use Protobuf to serialize and deserialize the data.

If there are any other I'd love to hear!